benches/state_space.rs

```diff
@@ -1,5 +1,104 @@
-use criterion as _;
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
 
-fn main() {
-    todo!("Implement benchmarks for `state_space` module!");
+fn criterion_benchmark(c: &mut Criterion) {
+    c.bench_function("add state changes benchmark", |b| {
+        b.iter(|| bench_funcs::test_add_state_changes(black_box(3)));
+    });
+    c.bench_function("add empty state changes benchmark", |b| {
+        b.iter(|| bench_funcs::test_add_empty_state_changes(black_box(3)));
+    });
+}
+
+criterion_group!(benches, criterion_benchmark);
+criterion_main!(benches);
+
+mod bench_funcs {
+    use std::sync::Arc;
+
+    use alloy::primitives::Address;
+    use amms::{
+        amm::{uniswap_v2::UniswapV2Pool, AMM},
+        state_space::{error::StateChangeError, StateChange, StateChangeCache},
+    };
+
+    use tokio::sync::RwLock;
+
+    /// Duplicated helper method entirely as in `state_space::tests` module
+    /// Reason of duplication: to retain encapsulation
+    async fn add_state_change_to_cache(
```

Comment:
Instead of adding this function, lets just mark the existing function in `src/state_space/mod.rs` as `pub`.